### PR TITLE
[relay-lsp] Support go to definition for arguments

### PR DESCRIPTION
**Pull request by tobias-tengler**

### CODE REVIEW

**Comment by captbaritone** [comment]
Exciting! Feel free to ping me when this PR is ready to review.

**Comment by tobias-tengler** [comment]
@captbaritone ready to review :)

File: compiler/crates/relay-lsp/src/goto_definition/mod.rs [hunk 1]
```diff
@@ -44,6 +44,15 @@ pub enum DefinitionDescription {
         parent_type: Type,
         field_name: StringKey,
     },
+    FieldArgument {
+        parent_type: Type,
+        field_name: StringKey,
+        argument_name: StringKey,
+    },
+    DirectiveArgument {
+        directive_name: StringKey,
```
Comment (captbaritone) [inline compiler/crates/relay-lsp/src/goto_definition/mod.rs:53]: Don't we have a [new type](https://doc.rust-lang.org/rust-by-example/generics/new_types.html) for directive name, and perhaps argument name? If so, could we use them here?

**Comment by captbaritone** [comment]
Looks good overall. Just one question (see inline comments) before I import.

**Pull request closed without merging**